Makefile

```diff
@@ -1,12 +1,42 @@
 current_dir := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+CARGO ?= $(shell which cargo)
+
+ifdef RUST_TARGET
+	TARGET = --target ${RUST_TARGET}
+endif
 
 build:
-	cargo build --features="fusedev"
-	cargo build --features="virtiofs"
-	cargo build --features="vhost-user-fs"
-	cargo build --features="fusedev,async-io"
-	cargo build --features="virtiofs,async-io"
-	cargo build --features="vhost-user-fs,async-io"
+	${CARGO} build ${TARGET} --features="fusedev"
+	${CARGO} build ${TARGET} --features="virtiofs"
+	${CARGO} build ${TARGET} --features="vhost-user-fs"
+	${CARGO} build ${TARGET} --features="fusedev,async-io"
+	${CARGO} build ${TARGET} --features="virtiofs,async-io"
+	${CARGO} build ${TARGET} --features="vhost-user-fs,async-io"
+
+check: build
+	${CARGO} fmt -- --check
+	${CARGO} clippy ${TARGET} --features="fusedev" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="virtiofs" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="vhost-user-fs" --no-default-features -- -Dwarnings
+	${CARGO} clippy ${TARGET} --features="fusedev,virtiofs" --no-default-features -- -Dwarnings
+
+test:
+	cargo test ${TARGET} --features="fusedev" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="virtiofs" --no-default-features  -- --nocapture --skip integration
+	cargo test ${TARGET} --features="vhost-user-fs" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="fusedev,virtiofs" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="fusedev,async-io" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="virtiofs,async-io" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="vhost-user-fs,async-io" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="fusedev,virtiofs,async-io" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --features="fusedev,persist" --no-default-features -- --nocapture --skip integration
+	cargo test ${TARGET} --all-features -- --nocapture --skip integration
+
+smoke:
+	cargo test ${TARGET} --features="fusedev,persist" -- --nocapture
+
+smoke-all: smoke
+	cargo test ${TARGET} --features="fusedev,persist" -- --nocapture --ignored
```

Comment:
Same here for "smoke" and "smoke-all" targets, use ${CARGO} instead of cargo ?

Reply:
Ditto here.